mods/sethome/init.lua

```diff
@@ -24,42 +25,58 @@ loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
+function sethome.go_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		-- just a check to prevent the server crashing
+		return false
+	end
+	if homepos[name] then
+		player:setpos(homepos[name])
+		minetest.chat_send_player(name, "Teleported to home!")
+		return true
+	else
+		minetest.chat_send_player(name, "Set a home using /sethome")
+		return false
+	end
+end
 
 minetest.register_chatcommand("home", {
     description = "Teleport you to your home point",
     privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+    func = sethome.go_home,
 })
 
+function sethome.set_home(name, pos)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		return false
+	end
+
+	if not pos or pos == "" then
+		pos = player:getpos()
```

Comment:
Depending on what will call the function, it can be both. If it is used as the callback for "/sethome" then param, the second parameter, here called pos, will be "" (or some other string, which I realize now is problematic). As suggested [here](https://github.com/minetest/minetest_game/pull/747#discussion_r48074571) if pos is nil/"" then the player's position is used.